commands.js

```diff
@@ -818,8 +820,10 @@ exports.commands = {
 			if (targetUser && Rooms.rooms[room.id].users[targetUser.userid]) targetUser.popup("You were demoted to Room " + groupName + " by " + user.name + ".");
 		} else if (nextGroup === '#') {
 			this.addModCommand("" + name + " was promoted to " + groupName + " by " + user.name + ".");
+			targetUser.popup("You were promoted to " + groupName + " by " + user.name + ".");
 		} else {
 			this.addModCommand("" + name + " was promoted to Room " + groupName + " by " + user.name + ".");
+			targetUser.popup("You were promoted to Room " + groupName + " by " + user.name + " in " + room + ".");
```

Comment:
You should check that the user is in the room to avoid issues with <code>/roomvoice</code>. Also same crash here as above.